lib/rspec/matchers.rb

```diff
@@ -745,6 +745,38 @@ def raise_error(error=Exception, message=nil, &block)
       desc.sub("raise", "raising")
     end
 
+    # Matches if an expectation fails
+    #
+    # @example
+    #
+    #   expect { some_expectation }.to fail
+    def fail
+      raise_error(RSpec::Expectations::ExpectationNotMetError)
+    end
+
+    # Matches if an expectation fails with the provided message
+    #
+    # @example
+    #
+    #   expect { some_expectation }.to fail_with("some failure message")
+    #   expect { some_expectation }.to fail_with(/some failure message/)
+    def fail_with(message)
+      raise_error(RSpec::Expectations::ExpectationNotMetError, message)
+    end
+
+    # Matches if an expectation fails including the provided message
+    #
+    # @example
+    #
+    #   expect { some_expectation }.to fail_including("some failure message")
+    #   expect { some_expectation }.to fail_including(/some failure message/)
```

Comment:
Passing a regex to `fail_including` doesn't make any sense.  Use `fail_with` if you're using a regex (as that will match properly), but the regex will not work here because a string does not _include_ a regex, it _matches_ one.  (OTOH, if the message was an array containing regex -- such as `[/some regex/]` -- then it could work, but it's not, so the regex example is nonsensical).